app/controllers/documents_controller.rb

```diff
@@ -101,7 +101,16 @@ def document_error_messages
   end
 
   def fetch_document
-    @document = document_klass.find(params[:content_id])
+    begin
+      @document = document_klass.find(params[:content_id])
+    rescue Document::RecordNotFound => e
+      flash[:danger] = "Document not found"
+      redirect_to documents_path(document_type: document_type)
```

Comment:
Fair enough - replicating existing behaviour seems legit.  FWIW I checked contacts-admin and imminence and they seem happy to 404 in admin pages, so maybe something to add to a board or whatever to check for a wider pattern and conform later.